shell/common/rasterizer.cc

```diff
@@ -279,23 +276,22 @@ std::pair<sk_sp<SkImage>, std::string> Rasterizer::MakeGpuImage(
   TRACE_EVENT0("flutter", "Rasterizer::MakeGpuImage");
   FML_DCHECK(display_list);
 
-  switch (gpu_image_behavior_) {
-    case MakeGpuImageBehavior::kGpu:
-      break;
-    case MakeGpuImageBehavior::kBitmap:
-      return MakeBitmapImage(picture_size);
-  }
+  const SkImageInfo image_info = SkImageInfo::MakeN32Premul(picture_size);
+  bool gpu_available = true;
+  delegate_.GetIsGpuDisabledSyncSwitch()->Execute(
+      fml::SyncSwitch::Handlers().SetIfTrue(
+          [&gpu_available] { gpu_available = false; }));
```

Comment:
Can we put the following logic into `SyncSwitch`'s `Handlers`? I guess just using `SyncSwitch` to get the value of `gpu_available` is not enough.

Reply:
Yeah I'll have to refactor this. I realized after uploading

Reply:
Done